tcms/testruns/static/testruns/js/get.js

```diff
@@ -1,5 +1,88 @@
+let executionStatuses;
+
 $(document).ready(() => {
 
     $('.bootstrap-switch').bootstrapSwitch();
 
+    const testRunId = $('#test_run_pk').data('pk')
+
+    jsonRPC('TestExecutionStatus.filter', {}, data => {
+        executionStatuses = data
+        drawPercentBar(testRunId)
+    })
 })
+
+function drawPercentBar(testRunId) {
+
+    jsonRPC('TestExecution.filter', { 'run_id': testRunId }, testExecutions => {
+
+        let positiveCount = 0;
+        let negativeCount = 0;
+        let allCount = testExecutions.length;
+        let statusCount = {}
+        executionStatuses.forEach(s => statusCount[s.name] = { count: 0, id: s.id })
+
+        testExecutions.forEach(testExecution => {
+            const executionStatus = executionStatuses.find(s => s.id === testExecution.status_id)
+
+            if (executionStatus.weight > 0) {
+                positiveCount++
+            } else if (executionStatus.weight < 0) {
+                negativeCount++
+            }
+
+            statusCount[executionStatus.name].count++
+        })
+
+        renderProgressBars(positiveCount, negativeCount, allCount)
+        renderCountPerStatusList(statusCount)
+    })
+}
+
+function renderProgressBars(positiveCount, negativeCount, allCount) {
+
+    const positivePercent = +(positiveCount / allCount * 100).toFixed(2)
+    const positiveBar = $(".progress > .progress-completed")
+    if (positivePercent) {
+        positiveBar.text(`${positivePercent}%`)
+    }
+    positiveBar.css('width', `${positivePercent}%`)
+    positiveBar.attr('title', `${positivePercent}% Completed`)
+    positiveBar.attr('aria-valuenow', `${positivePercent}`)
+
+    const negativePercent = +(negativeCount / allCount * 100).toFixed(2)
+    const negativeBar = $('.progress > .progress-failed')
+    if (negativePercent) {
+        negativeBar.text(`${negativePercent}%`)
+    }
+    negativeBar.css('width', `${negativePercent}%`)
+    negativeBar.attr('title', `${negativePercent}% Failed`)
+    negativeBar.attr('aria-valuenow', `${negativePercent}`)
+
+    const remainingPercent = 100 - (negativePercent + positivePercent)
```

Comment:
variable naming !

Reply:
what's wrong with that?

Reply:
It doesn't reflect reality. We've got positive, negative and neutral statuses, hence percentages are named in the same way. It doesn't help if we don't use our own terminology consistently.